### ISSUE TITLE
rumprun-invoked qemu crashes on mirageos rump kernels

### ISSUE THREAD

**Issue by yomimono** [mato/opam-rumprun#3]
Following the steps in the README, the "hello, world" and unixtime examples build and run with `rumprun`, but when attempting the console or stackv4 MirageOS examples, the results are less good:

```
=== calling "./mir-console.bin" main () ===

cannot handle page request order 2/0x0!
panic: uvm_km_alloc failed
rump kernel halting...
PANIC: rumpuser panic
halted
```

I'm not sure what diagnostic information would be most helpful here, so any general pointers on where to look next or how to enable more useful output would be appreciated.  The qemu command line invoked by `rumprun` is `qemu-system-i386 -net none -enable-kvm -cpu host    -m 64 -kernel ./mir-console.bin`.  `qemu-system-i386` is version 2.4.1 ; the host OS is Ubuntu LTS 14.04.3.

**Comment by mato** [comment]
`cannot handle page request order ...` means "Out of memory". The default is 64MB, can you try with `rumprun kvm -M 128...` or similar?

**Comment by yomimono** [comment]
I get the same results when substituting 128, 1024 or 2048 for 64 in the `rumprun` script.

**Comment by mato** [comment]
Can you post a full boot log please?

**Comment by mato** [comment]
Also, what's your qemu-system-x86_64 --version?

**Comment by yomimono** [comment]
What's actually invoked is `qemu-system-i386`, but the version of both that and `-x86_64` is 2.4.1.  Getting a bootlog shortly.

**Comment by yomimono** [comment]
[A previous release of rumprun](https://github.com/rumpkernel/rumprun/tree/7695cf4a153a6d54982bacd2848334a8d68d5595) doesn't have this problem (thanks @yallop for the detective work).

**Comment by yallop** [comment]
`git bisect` reports that the problem was introduced in https://github.com/rumpkernel/rumprun/commit/b30d26

**Comment by mato** [comment]
@anttikantee, any ideas? Looks like a bug introduced by the recent mem allocator changes.

**Comment by mato** [comment]
Noting another data point, standalone OCaml unikernels (e.g. hello and unixtime from the opam-rumprun howto) do _not_ trigger this problem for me, only the mirage unikernels.

**Comment by anttikantee** [comment]
Should be fixed in rumpkernel/rumprun@303031bad77ee84e2eac6b65543f9c8bc16b8b67

Would still be interesting to know what the backtrace for the failure is.

**Comment by yallop** [comment]
rumpkernel/rumprun@303031b fixes things for me.
